plugins/cpp_metrics/parser/src/cppmetricsparser.cpp

```diff
@@ -165,6 +168,170 @@ void CppMetricsParser::functionBumpyRoad()
   });
 }
 
+template <typename T>
+void CppMetricsParser::checkTypes(
+  const std::string path,
+  const std::unordered_set<std::uint64_t>& typesFound,
+  const std::unordered_map<std::uint64_t,std::string>& typeDefinitionPaths,
+  std::unordered_map<std::string,std::unordered_set<std::uint64_t>>& relationsFoundInFile,
+  int& relationsInModule
+)
```

Comment:
Remark: once different C++ code metrics have been separated into different classes, this should be refactored to use class fields instead of passing all these structures are references.